mllib/src/main/scala/org/apache/spark/mllib/fpm/LocalPrefixSpan.scala

```diff
@@ -42,22 +44,20 @@ private[fpm] object LocalPrefixSpan extends Logging with Serializable {
   def run(
       minCount: Long,
       maxPatternLength: Int,
-      prefix: Array[Int],
-      projectedDatabase: Array[Array[Int]]): Array[(Array[Int], Long)] = {
+      prefix: ArrayBuffer[Int],
```

Comment:
`ArrayBuilder` is better than `ArrayBuffer` for `Int`. The latter is not specialized for `Int` and hence has boxing/unboxing overhead. But here, we may want to consider `List` to avoid re-allocating buffers. The cost is that we have to inverse the list, e.g., https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/mllib/fpm/FPTree.scala#L72.